### DISCUSSION TITLE
Asterisk labels support

### DISCUSSION THREAD

**Discussion by cactusFriday** [PowerDNS/pdns#13394] [Q&A] [answered]
Hello!
I recently read the second section in [RFC 4592](https://datatracker.ietf.org/doc/html/rfc4592#section-2.1.2) which says that domain names like `asterisk.*.example.ru` are valid. PowerDNS currently only supports wildcard labels, i.e. asterisk at the beginning of the domain name (as the first label).
I see that there is a function in PowerDNS Auth code, which, if I understand correctly, prohibits asterisk everywhere but the first character. 
https://github.com/PowerDNS/pdns/blob/master/pdns/ws-api.cc#L398 

Is there any support planned for asterisk domain names? Or am I doing something wrong trying to add such record set?

```
PATCH example.com (PDNS API)
{"rrsets": [{"name": "asterisk.*.example.com", "ttl": 3600, "changetype": "REPLACE", "type": "A", "records": [{"content": "1.1.1.1", "disabled": False}]}]}
```
I'm using pdns-4.7.3 with mysql backend

**BUT**
Using the `pdnsutil`, I successfully created such record set
```
$ pdnsutil --config-name api-1 list-zone example.com
asterisk.*.example.com	3600	IN	A	1.1.1.1
example.com	3600	IN	SOA	a.misconfigured.dns.server.invalid hostmaster.example.com 0 10800 3600 604800 3600

mysql:
+------+-----------+------------------------+------+------------------------------------------------------------------------------------+------+------+----------+-----------+------+
| id   | domain_id | name                   | type | content                                                                            | ttl  | prio | disabled | ordername | auth |
+------+-----------+------------------------+------+------------------------------------------------------------------------------------+------+------+----------+-----------+------+
| 9026 |        12 | example.com            | SOA  | a.misconfigured.dns.server.invalid hostmaster.example.com 0 10800 3600 604800 3600 | 3600 |    0 |        0 | NULL      |    1 |
| 9027 |        12 | asterisk.*.example.com | A    | 1.1.1.1                                                                            | 3600 |    0 |        0 | NULL      |    1 |
+------+-----------+------------------------+------+------------------------------------------------------------------------------------+------+------+----------+-----------+------+
```
And resolve:
```
dig @localhost asterisk.*.example.com a +short
1.1.1.1
```

**Comment by Habbie** [comment]
What is the goal?

**Reply by cactusFriday** [reply]
My goal is to create record set with name `asterisk.*.example.com` using the PowerDNS API
When I try to do it, reply is 
```
 {'error': "Name 'asterisk.*.example.com.' contains unsupported characters"}
```

**Reply by Habbie** [reply]
ok, but why do you want to do that?

**Reply by cactusFriday** [reply]
I just simply want to create such record set as it's a valid and everything works fine)
I'm just curious, so I've came here to ask :) 
And I'm wondering why validation only allows wildcard

**Reply by Habbie** [reply, marked as answer]
The validation is stricter than the RFCs require, because we want to prevent common mistakes. Sometimes we will adjust the validation based on a goal a user fails to achieve under the current limitations. This does not sound like that :-)

**Reply by cactusFriday** [reply]
Thank you!
